security.md

```diff
@@ -0,0 +1,135 @@
+        Console.WriteLine($"C: {clientBlob}");
+        serverBlob = serverAuthentication.GetOutgoingBlob(clientBlob, out var serverStatusCode);
+        if (serverStatusCode != NegotiateAuthenticationStatusCode.Completed &&
+            serverStatusCode != NegotiateAuthenticationStatusCode.ContinueNeeded)
+        {
+            Console.WriteLine($"Server authentication failed with status code {serverStatusCode}");
+            break;
+        }
+        Console.WriteLine($"S: {serverBlob}");
+    }
+    else
+    {
+        Console.WriteLine(
+            clientStatusCode == NegotiateAuthenticationStatusCode.Completed ?
+            "Successfully authenticated" :
+            $"Authentication failed with status code {clientStatusCode}");
+        break;
+    }
+}
+```
+
+Once the authenticated session is established, the `NegotiateAuthentication` instance can be used to sign/encrypt the outgoing messages and verify/decrypt the incoming messages. This is done through the [Wrap](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.negotiateauthentication.wrap?view=net-7.0) and [Unwrap](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.negotiateauthentication.unwrap?view=net-7.0) methods.
+
+This change was done as well as this part of the article was written by a community contributor [@filipnavara](https://github.com/filipnavara).
+
+
+## Performance
+
+Most of the networking perfomance improvements in .NET 7 are covered by Stephen's article [Performance Improvements in .NET 7 - Networking](https://devblogs.microsoft.com/dotnet/performance_improvements_in_net_7/#networking), but some of the security ones are worth mentioning again.
+
+### TLS Resume
+
+Establishing new TLS connection is fairly expensive operation as it requires multiple steps and several roundtrips. In scenarios where connection to the same server is re-created very often, time consumed by the handshakes will add up. TLS offers feature to mitigate this called Session Resumption, see [RFC 5246 - 7.3.  Handshake Protocol Overview](https://www.rfc-editor.org/rfc/rfc5246.html#section-7.3) and [RFC 8446 - 2.2.  Resumption and Pre-Shared Key](https://www.rfc-editor.org/rfc/rfc8446#section-2.2). Even though the mechanics differ for different TLS versions, the end goal is the same, save a round-trip and some CPU time when re-establishing connection to a previously connected server. This feature is automatically provided by SChannel on Windows, but with OpenSSL on Linux it required several changes to enable this:
+- Server side (stateless): [PR #57079](https://github.com/dotnet/runtime/pull/57079) and [PR #63030](https://github.com/dotnet/runtime/pull/63030)
+- Client side: [PR #64369](https://github.com/dotnet/runtime/pull/64369)
+- Cache size control: [PR #69065](https://github.com/dotnet/runtime/pull/69065)
+
+These changes bring .NET 7 Linux support of TLS resume on par with Windows capabilities.
```

Comment:
There are still some differences. On Windows, the handshake is handled by separate process so the Resume can be shared among all processes on the system. For server the linux implementation is linked to SslStreamCertificate context and it will not work if the server side uses only certificate. 
It may be worth of mentioning DOTNET_SYSTEM_NET_SECURITY_DISABLETLSRESUME. While people may find it when reading the PRs but that may be too much for many.

Reply:
Added